### PR TITLE
Implement Login using oneauth

### LINKED ISSUE #2: Add: User authentication via oneauth
Allow user authentication using oneauth. 

- `/login` and `/logout` endpoints using oneauth
- '/users/me` to get details of current user.

Can you use either JWT or cookies for session management.

### PR DESCRIPTION
**Pull request by bhavyaagg**
Signed-off-by: Bhavya Aggarwal <bhavyaaggarwal1209@gmail.com>

I have implemented the login using passport-oneauth and Bearer Token

@championswimmer @abhishek97 Please have a look. 

Fixes #2

### CODE REVIEW

File: util/dbmodels.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Created by bhavyaagg on 23/05/18.
+ */
+
+const Sequelize = require('sequelize');
+const U = require('./util')
+const config = require('../config')
+
+
+const sequelize = new Sequelize(config.db.DB, config.db.USER, config.db.PASSWORD, {
```
Comment (abhishek97) [inline util/dbmodels.js]: Use `sequelize-cli` to generate models and migrations.

**Comment by bhavyaagg** [comment]
@abhishek97 I have updated my PR and used sequelize-cli.

File: config.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (abhishek97) [inline config.js]: Too many config files already. Use the one defined by `sequelize-cli`. For secrets, use ENV variables.

File: routes/users.js [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+router.get('/logout', function (req, res) {
+  req.logout();
+  res.redirect('/');
+});
+
+router.get('/me', passport.authenticate('bearer'), (req, res, next) => {
+  if (!req.user) {
+    return res.status(401).send("Please Login")
+  } else {
+    models.User.findOne({
```
Comment (abhishek97) [inline routes/users.js]: Can't we just `res.json(req.user)` here?

File: routes/users.js [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+});
+
+router.get('/logout', function (req, res) {
+  req.logout();
+  res.redirect('/');
+});
+
+router.get('/me', passport.authenticate('bearer'), (req, res, next) => {
+  if (!req.user) {
+    return res.status(401).send("Please Login")
```
Comment (abhishek97) [inline routes/users.js]: Have the `bearer` strategy return a `401` if we cannot authenticate the user.

File: config.js (hunk 2 above)
Comment (abhishek97) [reply config.js]: Also, a lot of code in this file seems irrelevant.

File: routes/users.js [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
-router.get('/', function(req, res, next) {
+router.get('/', function (req, res, next) {
   res.send('respond with a resource');
 });
 
+router.get('/login/cb', passport.authenticate('oneauth'));
+
+router.get('/login/cb/callback', passport.authenticate('oneauth', {failureRedirect: '/'}), function (req, res) {
+
+  return res.redirect('/?clienttoken=' + req.user.clienttoken);
```
Comment (abhishek97) [inline routes/users.js]: Can you explain more about this?
Why redirect to `/?clienttoken=XX` 

Also, I think you are mistaken here. We require Explicit Oauth2 flow here. We won't be having a `callback` handler on the server. The frontend will call the backend with the received `grant_code` and the backend just need to exchange that with `access-token`.

This is different from how auth works in `boss-backend`. You can take an example  (frontend part only) from codingblocks online and hackerblocks.

**Pull request closed without merging**